src/network/core/packet.cpp

```diff
@@ -370,6 +369,22 @@ std::vector<byte> Packet::Recv_buffer()
 	return data;
 }
 
+/**
+ * Extract at most as many bytes from the packet into the span.
```

Comment:
nitpick: this is a hard to read sentence. 

```suggestion
 * Extract at most the length of the span bytes from the packet into the span.
```

?